transcom/mymove
# B 19139 Fix long line of accounting not populating

## [B-19139](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A922317)

## Summary

Invoices were not generating properly due to previously hard lookups for “loa id”, which is our internal DB UUID. TACs and LOAs do not get tied together by this, this is a leftover from the manual import method Truss had conducted in the past. By updating to loa_sys_id, we’re able to successfully grab the TRDM UUID.

### How to test

1. Insert TAC and LOA scripts (See below) into your dev db
2. Login to office as service counselor
3. Select a move with an HHG shipment, make note of its locator #
4. Edit orders -> update TAC to be “TEST”
5. Edit orders -> update date issued to be ’24 Mar 2024’
6. Submit move details.
7. Login as TOO and approve HHG shipment
8. Login as prime -> navigate to move
9. Create a payment request on your HHG shipment for fuel surcharge
10. Login as TIO -> navigate to move
11. Approve payment request
12. Open browser dev tools -> navigate to network tab -> refresh the page
13. Select `payment-requests` network request
14. Grab your ID for the payment request you just made
15. Open postman
16. Import support.yaml
17. Select payment-requests -> paymentRequestID -> edit -> getPaymentRequestEDI
18. Provide this URL (Placing in your payment request ID)
  a. `https://primelocal:9443/support/v1/payment-requests/${payment_request_id}/edi`
19. Verify in your response that you see a FA2* section for the following. You can find these by searching your response for, eg, “FA2\*A1”, “FA2\*A3”

---BREAK---

1. A1
2. A3
3. A4
4. A5
5. B1
6. B2
7. C1
8. F1
9. I1
10. J1
11. L1
12. N1

<details>
<summary>TAC SQL</summary>

```
INSERT INTO transportation_accounting_codes (id, tac, loa_id, tac_sys_id, loa_sys_id, tac_fy_txt, tac_fn_bl_mod_cd, org_grp_dfas_cd, tac_mvt_dsg_id, tac_ty_cd, tac_use_cd, tac_maj_clmt_id, tac_bill_act_txt, tac_cost_ctr_nm, buic, tac_hist_cd, tac_stat_cd, trnsprtn_acnt_tx, trnsprtn_acnt_bgn_dt, trnsprtn_acnt_end_dt, dd_actvty_adrs_id, tac_blld_add_frst_ln_tx, tac_blld_add_scnd_ln_tx, tac_blld_add_thrd_ln_tx, tac_blld_add_frth_ln_tx, tac_fnct_poc_nm) 
VALUES 
('ab4e5115-5604-494e-afe8-5d86e9ba1911', 'TEST', NULL, 3422646, 28184127, '2024', '1', 'DZ', NULL, 'H', 'H', NULL, NULL, NULL, NULL, NULL, 'A', 'HHG TAC FOR OFFICER', '2023-10-01', '2024-09-30', NULL, NULL, NULL, NULL, NULL, 'CACI TEST'),
('ab4e5115-5604-494e-afe8-5d86e9ba1912', 'TEST', NULL, 3422647, 28184127, '2024', '3', 'DZ', NULL, 'H', 'H', NULL, NULL, NULL, NULL, NULL, 'A', 'HHG TAC FOR OFFICER', '2023-10-01', '2024-09-30', NULL, NULL, NULL, NULL, NULL, 'CACI TEST'),
('ab4e5115-5604-494e-afe8-5d86e9ba1913', 'TEST', NULL, 3422648, 28184127, '2024', '5', 'DZ', NULL, 'H', 'H', NULL, NULL, NULL, NULL, NULL, 'A', 'HHG TAC FOR OFFICER', '2023-10-01', '2024-09-30', NULL, NULL, NULL, NULL, NULL, 'CACI TEST'),
('ab4e5115-5604-494e-afe8-5d86e9ba1914', 'TEST', NULL, 3422649, 28184127, '2024', 'M', 'DZ', NULL, 'H', 'H', NULL, NULL, NULL, NULL, NULL, 'A', 'HHG TAC FOR OFFICER', '2023-10-01', '2024-09-30', NULL, NULL, NULL, NULL, NULL, 'CACI TEST'),
('ab4e5115-5604-494e-afe8-5d86e9ba1915', 'TEST', NULL, 3422650, 28184127, '2024', 'P', 'DZ', NULL, 'H', 'H', NULL, NULL, NULL, NULL, NULL, 'A', 'HHG TAC FOR OFFICER', '2023-10-01', '2024-09-30', NULL, NULL, NULL, NULL, NULL, 'CACI TEST');
```
</details>
<details>
<summary>LOA SQL</summary>

```
INSERT INTO lines_of_accounting (id, loa_sys_id, loa_dpt_id, loa_baf_id, loa_trsy_sfx_tx, loa_op_agncy_id, loa_allt_sn_id, loa_pgm_elmnt_id, loa_obj_cls_id, loa_bdgt_acnt_cls_nm, loa_doc_id, loa_instl_acntg_act_id, loa_dsc_tx, loa_bgn_dt, loa_end_dt, loa_fnct_prs_nm, loa_stat_cd, loa_hs_gds_cd, org_grp_dfas_cd, loa_trnsn_id, loa_end_fy_tx, loa_bg_fy_tx, created_at, updated_at) 
VALUES 
('ab4e5115-5604-494e-afe8-5d86e9ba1976', 28184127, 21, '0000', '0000', '01', '0000', '00000000', 'ABCD', 'ABCDEF', 'REF DOCS', '000000', 'HHG SHIPMENT FOR OFFICER', '2023-10-01', '2029-09-30', 'CACI TEST', 'U', 'HO', 'DZ', 'MP', '2024', '2024', '2024-03-25 12:00:00', '2024-03-25 12:00:00');
```
</details>
